### DISCUSSION TITLE
can Frigate receive as well as send MQTT ?

### DISCUSSION THREAD

**Discussion by ElectronicBattle** [blakeblackshear/frigate#13654] [Ask A Question] [answered]
Hello all

Are any of the following possible or (potentially) in the pipeline for future releases?

1 - I'd like Frigate to start recording a specific camera stream when an external MQTT event from an electrical switch occurs.  

2 - as per (1) but with a number of  seconds of pre- and post- detect (pre-roll and post-roll surrounding the event).

3 - Id like Frigate to "stamp" the existing timeline when an MQTT message is seen.  

The use cases for (1) cover events which (e.g.) cause a gate to open and later close, but the movement associated with the gate  is insufficient to reliably trigger Frigate, or the camera's view is obscured.  The second opportunity is merely to better guarantee that actions leading and following an event are caught.  The third option is so that you can have a timeline marked with an MQTT event ("gate opened" "gate closed" "lights on" "lights off" etc.).

I have an MQTT broker running c/o Home Assistant (Frigate is standalone) so I understand the idea of pub/sub but it isn't clear to me whether Frigate simply sends events (dog detected on camera three).

**Comment by NickM-27** [comment, marked as answer]
1. yes, the mqtt docs cover this, you can control many things with MQTT. All of this is available in the frigate integration in HA already. 
2. if you want to create an event on the timeline you must use the http api

**Reply by ElectronicBattle** [reply]
Thanks - I know where to start looking!
